action/proposer_register.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package action
+
+import (
+	"math/big"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-core/pkg/version"
+	"github.com/pkg/errors"
+)
+
+const (
+	// ProposerRegisterPayloadGas is the gas to register a proposer
+	ProposerRegisterPayloadGas = uint64(100)
+	// ProposerRegisterBaseIntrinsicGas is the base intrinsic gas to register a proposer
+	ProposerRegisterBaseIntrinsicGas = uint64(10000)
+	// ProposerRegisterAmount is the amount to register a proposer
+	ProposerRegisterAmount = uint64(10000)
```

Comment:
you will need to add an ABI interface for handling web3 format tx, refer to https://github.com/iotexproject/iotex-core/pull/3094
or maybe can be in next PR